stl/inc/functional

```diff
@@ -1207,6 +1207,604 @@ _NODISCARD bool operator!=(nullptr_t, const function<_Fty>& _Other) noexcept {
+                } else {
+                    _Impl._Destroy = _Function_deallocate_large_overaligned<alignof(_Vt)>;
+                }
+            } else {
+                _Impl._Destroy = _Function_destroy_large<_Vt>;
+            }
+        } else {
+            _Impl._Invoke = _Function_invoke_small<_Vt, _VtInvQuals, _Rx, _Types...>;
+
+            if constexpr (is_trivially_copyable_v<_Vt> && is_trivially_destructible_v<_Vt>) {
+                if constexpr ((_Function_small_copy_size<_Vt>) > sizeof(void*) * 2) {
+                    _Impl._Move = _Function_move_memcpy<_Function_small_copy_size<_Vt>>;
+                } else {
+                    _Impl._Move = nullptr;
+                }
+            } else {
+                _Impl._Move = _Function_move_small<_Vt>;
+            }
+
+            if constexpr (is_trivially_destructible_v<_Vt>) {
+                _Impl._Destroy = nullptr;
+            } else {
+                _Impl._Destroy = _Function_destroy_small<_Vt>;
+            }
+        }
+        return _Impl;
+    }
+
+    template <class _Vt, class _VtInvQuals>
+    _NODISCARD static const _Impl_t* _Create_impl_ptr() noexcept {
+        static constexpr _Impl_t _Impl = _Create_impl<_Vt, _VtInvQuals>();
+        return &_Impl;
+    }
+};
+
+template <class _Fn>
+class _Move_only_function_call {
+    static_assert(_Always_false<_Fn>, "std::move_only_function only accepts function types as template arguments, "
+                                      "with possibly const/ref/noexcept qualifiers.");
+};
+
+template <class _Rx, class... _Types>
+class _Move_only_function_call<_Rx(_Types...)> : public _Move_only_function_base<_Rx, _Types...> {
+public:
+    using result_type = _Rx;
+
+    template <class _Vt>
+    using _VtInvQuals = _Vt&;
+
+    template <class _Vt>
+    static constexpr bool _Is_callable_from =
+        is_invocable_r_v<_Rx, _Vt, _Types...>&& is_invocable_r_v<_Rx, _Vt&, _Types...>;
+
+    _Rx operator()(_Types... _Args) {
+        return this->_Get_invoke()(this->_Data, _STD forward<_Types>(_Args)...);
+    }
+};
+
+template <class _Rx, class... _Types>
+class _Move_only_function_call<_Rx(_Types...)&> : public _Move_only_function_base<_Rx, _Types...> {
+public:
+    using result_type = _Rx;
+
+    template <class _Vt>
+    using _VtInvQuals = _Vt&;
+
+    template <class _Vt>
+    static constexpr bool _Is_callable_from = is_invocable_r_v<_Rx, _Vt&, _Types...>;
+
+    _Rx operator()(_Types... _Args) & {
+        return this->_Get_invoke()(this->_Data, _STD forward<_Types>(_Args)...);
+    }
+};
+
+template <class _Rx, class... _Types>
+class _Move_only_function_call<_Rx(_Types...) &&> : public _Move_only_function_base<_Rx, _Types...> {
+public:
+    using result_type = _Rx;
+
+    template <class _Vt>
+    using _VtInvQuals = _Vt&&;
+
+    template <class _Vt>
+    static constexpr bool _Is_callable_from = is_invocable_r_v<_Rx, _Vt&&, _Types...>;

```

Comment:
They ask specifically for copying _inv-quals_ [[func.wrap.move.ctor]/1](https://eel.is/c++draft/func.wrap.move#ctor-1),  _inv-quals_ are [[func.wrap.move.general]/2](https://eel.is/c++draft/func.wrap.move#general-2)

I'll change that, but then it would be spelled differently from what we are asked for.

Reply:
Understood - the Standard itself also generally avoids the `&&`, this is just a rare case where it appears to be implied. I think the difference is acceptable.